### PR TITLE
error format

### PR DESCRIPTION
**Pull request by jonnyalexbh**
## Summary
- error handling in GraphQL

## Trello Card
https://trello.com/c/cRweKkew/19-manejo-de-errores-en-graphql

## Screenshot
Screen Shot 2019-08-20 at 10 43 21 AM

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -14,9 +14,17 @@ migrationsManager
...
     }); */
-    new ApolloServer({ schema }).listen(port).then(({ url, subscriptionsUrl }) => {
-      logger.info(`🚀 Server ready at ${url}`);
-      logger.info(`🚀 Subscriptions ready at ${subscriptionsUrl}`);
+    new ApolloServer({
+      schema,
+      formatError: err => ({
+        message: err.message,
+        statusCode: err.extensions.code
+      })
```
Comment (emanuelcasco) [inline server.js]: - Abstract `formatError` to a function, reference it here. 
- Make sure to log errors in the function.
- WRITE TEST! This is very important, you should write test for every new functionality.

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: server.js (hunk 1 above)
Comment (emanuelcasco) [reply server.js]: Also, make sure to catch errors in every service/resolver you are using. All errors should be catched and parsed before pass it to Apollo. For example:

- Return 404 when required resource is not available.
- Return 400 when input is invalid.
- Return 503 when database connection fails.
- And so on...

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

**Review by emanuelcasco — APPROVED** [review]

File: app/services/album.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 const axios = require('axios'),
   config = require('../../config'),
   base_uri = config.common.albumsApi.url,
+  error = require('../errors'),
```
Comment (emanuelcasco) [inline app/services/album.js]: Name it errors

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: server.js [hunk 3]
```diff
@@ -14,9 +15,14 @@ migrationsManager
       enabled: !!config.common.rollbar.accessToken,
       environment: config.common.rollbar.environment || config.environment
     }); */
-    new ApolloServer({ schema }).listen(port).then(({ url, subscriptionsUrl }) => {
-      logger.info(`🚀 Server ready at ${url}`);
-      logger.info(`🚀 Subscriptions ready at ${subscriptionsUrl}`);
+    new ApolloServer({
+      schema,
+      formatError: err => helpers.showError(err)
```
Comment (emanuelcasco) [inline server.js]: Replace for `formatError : helpers.showError`

File: app/models/user.js [hunk 4]
```diff
@@ -29,15 +32,35 @@ module.exports = (sequelize, DataTypes) => {
...
+      logger.error(err);
+      throw errors.databaseError(err);
+    });
 
-  User.prototype.updateModel = props => this.update(props);
+  User.prototype.updateModel = props =>
+    this.update(props).catch(err => {
+      logger.error(err);
+      throw errors.databaseError(err);
+    });
```
Comment (emanuelcasco) [inline app/models/user.js:63]: Do you test this method? Where are you using it?

**Review by emanuelcasco — CHANGES_REQUESTED** [review]

File: app/models/user.js (hunk 4 above)
Comment (jonnyalexbh) [reply app/models/user.js:63]: this method was by default in the bootstrap

Comment (emanuelcasco) [reply app/models/user.js:63]: Ok, no problem. But take this in mind: https://blog.pragmatists.com/the-many-faces-of-this-in-javascript-5f8be40df52e

**Review by emanuelcasco — APPROVED** [review]
Great work here! 

image

**Pull request merged**